### PR TITLE
Redirect user to new page after deleting page

### PR DESCRIPTION
**Pull request by Devorein**
[Task Card](https://app.charmverse.io/charmverse/page-49366552253645923?viewId=d15c9b90-8918-44da-bbd8-db8712faa723&cardId=a479fff9-c4e4-4dc5-ad67-5cdd5ad0c37a)

### CODE REVIEW

File: hooks/usePages.tsx [hunk 1]
```diff
@@ -135,7 +135,7 @@ export function PagesProvider ({ children }: { children: ReactNode }) {
 
     return computedPermissions;
   }
-
+  // eslint-disable-next-line react/no-unused-prop-types
```
Comment (msadura) [inline hooks/usePages.tsx:138]: why is this needed?

File: components/common/PageLayout/components/PageNavigation/components/PageTreeItem.tsx [hunk 2]
```diff
@@ -394,11 +396,16 @@ function PageActionsMenu ({ closeMenu, pageId, pagePath }: { closeMenu: () => vo
     }
     const page = pages[pageId];
     const board = boards.find(b => b.id === page?.id);
-
-    await deletePage({
+    const currentPage = pages[currentPageId];
+    const newPage = await deletePage({
```
Comment (msadura) [inline components/common/PageLayout/components/PageNavigation/components/PageTreeItem.tsx:400]: `deletePage` action returns `newPage` - conceptually this feels really weird 🤯

File: hooks/usePages.tsx (hunk 1 above)
Comment (Devorein) [reply hooks/usePages.tsx:138]: I have no idea why vscode was throwing this random error on that line. Tried restarting ts server and even vscode, but nothing changed. Let me try it out again as see if it persists, if not I will remove it.

Comment (Devorein) [reply hooks/usePages.tsx:138]: Fixed the issue. Thnx for the suggestion :D

**Review by msadura — APPROVED** [review]
